packages/react-shopify-app-route-propagator/src/index.ts

```diff
@@ -15,58 +15,38 @@ export interface Props {
 
 export const MODAL_IFRAME_NAME = 'app-modal-iframe';
 
-export default class ReactShopifyAppRoutePropagator extends React.PureComponent<
-  Props,
-  never
-> {
-  private historyInstance: History.History | undefined;
+export default function ReactShopifyAppRoutePropagator({app, location}: Props) {
```

Comment:
Can this just be a hook instead? You can keep a component (that just calls the hook and returns `null`), but a custom hook here would probably be nice for folks

Reply:
API-wise, it is weird to me that you need to manually pass in `location`, I am surprised it does not just come in through a hook

Reply:
Now might be the right time to change this since it is already a breaking release.

Reply:
do we want to have dependency on react router? I am scare because the v4 -> v5 fiasco